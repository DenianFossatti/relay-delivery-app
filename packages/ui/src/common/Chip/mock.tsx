import { ChipProps } from '.';

export const mockChips: ChipProps = {
  id: '1232',
  isSelected: false,
  label: 'Banana',
  onPress: () => null,
  testID: 'Chip',
};
