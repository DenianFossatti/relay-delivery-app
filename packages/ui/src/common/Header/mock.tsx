import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';

import { HStack } from '../HStack';

import { HeaderProps } from './';

type headerVariants =
  | 'CenteredLogo'
  | 'Title'
  | 'TitleAndSubtitle'
  | 'CustomSubtitleComponent'
  | 'leftButtons'
  | 'rightButtons'
  | 'menuProps'
  | 'NewsFeed'
  | 'WebViewForm';

export const mockHeaderRightButons: HeaderProps['rightButtons'] = [
  {
    accessibilityLabel: 'Ajuda',
    icon: ['far', 'question-circle'],
    onPress: () => console.log('ajuda'),
  },
];

export const mockHeader: Record<headerVariants, HeaderProps> = {
  CenteredLogo: {
    logoCenter: true,
  },
  CustomSubtitleComponent: {
    subtitle: (
      <HStack alignItems="center" space={1}>
        <FontAwesomeIcon color="#00AD8E" icon="lock" size={14} />
        Page Subtitle
      </HStack>
    ),
    title: 'Page Title',
  },
  NewsFeed: {
    logo: true,
    logoCenter: false,
    menuProps: {
      items: [
        { onPress: () => console.log('onPress'), title: 'Menu Option 1' },
        { icon: 'home', onPress: () => console.log('onPress'), title: 'Home' },
        {
          icon: 'arrow-right-from-bracket',
          onPress: () => console.log('onPress'),
          title: 'Logout',
        },
      ],
      triggerIconProps: {
        icon: ['far', 'user'],
      },
    },
    rightButtons: [
      {
        accessibilityLabel: 'User',
        icon: ['far', 'eye'],
        onPress: () => console.log('user'),
      },
      {
        accessibilityLabel: 'Question',
        icon: ['far', 'question-circle'],
        onPress: () => console.log('bell'),
      },
    ],
  },
  Title: {
    title: 'Page Title',
  },
  TitleAndSubtitle: {
    subtitle: 'Page Subtitle',
    title: 'Page Title',
  },
  WebViewForm: {
    leftButtons: [
      {
        accessibilityLabel: 'WebView',
        icon: 'times',
        onPress: () => console.log('onPress'),
      },
    ],
    menuProps: {
      items: [
        {
          icon: ['fas', 'share-nodes'],
          onPress: () => console.log('onPress'),
          title: 'Compartilhar',
        },
        {
          icon: ['far', 'copy'],
          onPress: () => console.log('onPress'),
          title: 'Copiar link',
        },
        {
          icon: ['far', 'window-maximize'],
          onPress: () => console.log('onPress'),
          title: 'Abrir no navegador',
        },
      ],
      triggerIconProps: {
        icon: ['fas', 'ellipsis'],
      },
    },
    subtitle: (
      <HStack alignItems="center" space={1}>
        <FontAwesomeIcon color="#00AD8E" icon="lock" size={14} />
        Page Subtitle
      </HStack>
    ),
    title: 'Page Title',
  },
  leftButtons: {
    leftButtons: [
      {
        accessibilityLabel: 'User',
        icon: 'times',
        onPress: () => console.log('onPress'),
      },
      {
        accessibilityLabel: 'User',
        icon: 'home',
        onPress: () => console.log('onPress'),
      },
    ],
    title: 'Page Title',
  },
  menuProps: {
    menuProps: {
      items: [
        { onPress: () => console.log('onPress'), title: 'Menu Option 1' },
        { icon: 'home', onPress: () => console.log('onPress'), title: 'Home' },
        {
          icon: 'arrow-right-from-bracket',
          onPress: () => console.log('onPress'),
          title: 'Logout',
        },
      ],
    },
    rightButtons: [
      {
        accessibilityLabel: 'User',
        icon: 'times',
        onPress: () => console.log('onPress'),
      },
    ],
    title: 'Page Title',
  },
  rightButtons: {
    rightButtons: [
      {
        accessibilityLabel: 'User',
        icon: 'times',
        onPress: () => console.log('onPress'),
      },
      {
        accessibilityLabel: 'User',
        icon: 'home',
        onPress: () => console.log('onPress'),
      },
    ],
    title: 'Page Title',
  },
};
