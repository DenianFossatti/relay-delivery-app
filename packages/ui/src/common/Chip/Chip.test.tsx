import { fireEvent, mockTestID, render, wrapper } from '../../tests';

import { mockChips } from './mock';

import { Chip, ChipProps } from '.';

describe(`Chips`, () => {
  const props: ChipProps = {
    ...mockChips,
  };

  const el_chip = mockTestID('Pressable', props.testID!);
  const el_chipText = mockTestID('Text', props.testID!);

  it(`should press on #${el_chip}`, async () => {
    const { findByTestId, toJSON } = render(<Chip {...props} />, {
      wrapper,
    });
    const sut = await findByTestId(el_chip);
    fireEvent.press(sut);
    expect(sut).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
  it(`should text on #${el_chipText}`, async () => {
    const { findByTestId, toJSON } = render(<Chip {...props} />, {
      wrapper,
    });
    const sut = await findByTestId(el_chipText);
    expect(sut.children.join('')).toEqual(props.label);
    expect(toJSON()).toMatchSnapshot();
  });
});
