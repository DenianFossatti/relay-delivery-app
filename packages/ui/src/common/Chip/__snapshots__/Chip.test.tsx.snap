// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Chips should press on #Pressable_Chip 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={false}
    collapsable={false}
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FE35F1",
        "borderRadius": 50,
        "borderWidth": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      }
    }
    testID="Pressable_Chip"
  >
    <Text
      accessible={false}
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": "Quicksand-Regular",
          "fontSize": 14,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0.25,
          "lineHeight": 17.5,
          "textDecorationLine": undefined,
        }
      }
      testID="Text_Chip"
    >
      Banana
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Atoms/Chips should text on #Text_Chip 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={false}
    collapsable={false}
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FE35F1",
        "borderRadius": 50,
        "borderWidth": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      }
    }
    testID="Pressable_Chip"
  >
    <Text
      accessible={false}
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": "Quicksand-Regular",
          "fontSize": 14,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0.25,
          "lineHeight": 17.5,
          "textDecorationLine": undefined,
        }
      }
      testID="Text_Chip"
    >
      Banana
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
